src/costs.jl

```diff
@@ -6,7 +6,15 @@ else
 end
 
 
+"""
+    Instruction
 
+`Instruction` is like a `GlobalRef` except it isn't. (Is this because you need it to be
+encodable in a type parameter? How do you handle submodules? Would making `mod` the result
+of `fullname` fix that?)
```

Comment:
Good suggestion. We could probably replace it with `GlobalRef`s.
We can't encode `Instruction` in a type parameter either, on a couple day old Julia master: 
```julia
julia> Val((:a, :b))
Val{(:a, :b)}()

julia> using LoopVectorization: Instruction

julia> Val(Instruction(:foo))
ERROR: TypeError: in Type, in parameter, expected Type, got a value of type Instruction
Stacktrace:
 [1] Val(::Instruction) at ./essentials.jl:696
 [2] top-level scope at REPL[13]:1
```
(condense/reconstruct)_loopset.jl use a vector of symbols.

I think this entire part of the code can be changed.
If you look at the `Base.Expr(::Instruction, args...)` method defined a few lines lower, you'll notice it more or less ignores the module; it'll only using a `Expr(:(.), mod, f)`/`GlobalRef` if the module is `LoopVectorization`.

This is because it's (now) supposed to work through one of the following means:
1) LoopVectorization "recognizes" the function, and it's defined in the LoopVectorization module's scope. Then it'll lower the function `f` as `LoopVectorization.f`
2) It will pass the function as an argument to `_avx_!`, and there will be a `gensym`ed name. Because the function has to be called within `_avx_!` within LoopVectorization, this is the best solution I'm aware of to make it inferrable and avoid world-age problems.

That wasn't the original design. `Instruction` could probably be replaced with a `Symbol`.